src/fw/services/normal/comm_session/app_session_capabilities.c

```diff
@@ -89,17 +97,35 @@ static void prv_rewrite_cb(SettingsFile *old_file,
 }
 
 void comm_session_app_session_capabilities_evict(const Uuid *app_uuid) {
+  mutex_lock(s_app_comm_mutex);
+
   SettingsFile settings_file;
   if (PASSED(prv_open(&settings_file))) {
     settings_file_delete(&settings_file, app_uuid, sizeof(*app_uuid));
     settings_file_close(&settings_file);
   }
+
+  mutex_unlock(s_app_comm_mutex);
 }
 
 void comm_session_app_session_capabilities_init(void) {
+  s_app_comm_mutex = mutex_create();
+
+  mutex_lock(s_app_comm_mutex);
```

Comment:
I guess it is safe to not lock here; you're just creating the mutex so any concurrent access at this point would be racey anyway.